### ISSUE TITLE
Pre-commit hook for check and lock not working in subdirectories.

### ISSUE THREAD

**Issue by gnkow** [python-poetry/poetry#7247]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
I've a multi-root project directory such as:

```
/api (poetry/python)
     /pyproject.toml
/api2 (non-python)
.pre-commit-config.yml
```

My .pre-commit-config.yml states:
```
repos:
  - repo: https://github.com/python-poetry/poetry
    rev: 1.3.1
    hooks:
      - id: poetry-check
      - id: poetry-lock
```

Whenever I try to use the pre-commit hook I get this:

```
poetry-check.........................................(no files to check)Skipped
poetry-lock..............................................................Failed
- hook id: poetry-lock
- exit code: 1

Poetry could not find a pyproject.toml file in <ROOT> or its parents
```

Is there a way to run the pre-commit from the `api` folder?

**Comment by gnkow** [comment]
Closing due to duplicate https://github.com/python-poetry/poetry/issues/7239

**Comment by rubenvdg** [comment]
Actually, this issue rightfully mentions that `poetry-lock` hook doesn't work if your poetry project is not in the root of your project and https://github.com/python-poetry/poetry/issues/7239 does not mention that. Therefore, I think this is not a duplicate (but a superset of https://github.com/python-poetry/poetry/issues/7239).

Maybe it'd be good to re-open?

**Comment by xinbinhuang** [comment]
I think the problem is that the current the `.pre-commit-config.yaml` hardcoded the `files: ^poetry.lock$`, which means that it'll only trigger the hook when the files are in the root directory. I created a PR to fix it: #7247
